### ISSUE TITLE
[Feature Request] Adding Minitest as a default debug configuration

### ISSUE THREAD

**Issue by Syntaf** [rubyide/vscode-ruby#630]
👋 any interest in adding a default minitest debug configuration? I can happily submit a PR if so.

After spending way too much time looking online for a solution for debugging failing Mintest tests in VSCode, I realized it was actually quite simple using this extension and the following config:

```json
    {
      "cwd": "${workspaceRoot}",
      "name": "Minitest - Run at cursor position",
      "type": "Ruby",
      "request": "launch",
      "program": "${workspaceRoot}/bin/rails",
      "args": [
        "test",
        "${file}:${lineNumber}"
      ]
    }
```

Since RSpec is an available default configuration it would be great to also have a Minitest one.

**Comment by wingrunr21** [comment]
Sure, please open a PR!

**Comment by nadnoslen** [comment]
Hey @Syntaf,

I have been trying that same configuration in my RoR 6 app.

Are you running a Rails app and if so what version?

When triggering the run configuration, my _Debug Console_ shows all the `rails test` dots, but they do not appear to halt for any of my breakpoints.  I wonder if I'm missing something.

I can confirm that my `rails server` debug configuration hits my breakpoints...I just really wanted my tests to be debuggable.

Any help you can offer would be awesome!

**Comment by Syntaf** [comment]
Do you have parallelization enabled for your tests in `test_helper.rb` @nadnoslen ? That might be the issue, I don't think VSCode Debugging plays well with parallelized tests

**Comment by nadnoslen** [comment]
Hey @Syntaf,

Thanks so much for replying!  I should have mentioned that I had played around with the `PARALLEL_WORKERS` environment variable:

```json
      {
        "args": ["test", "${file}:${lineNumber}"],
        "cwd": "${workspaceRoot}",
        "env": {
          "PARALLEL_WORKERS": 1
        },
        "name": "Minitest - Run at cursor position",
        "program": "${workspaceRoot}/bin/rails",
        "request": "launch",
        "type": "Ruby"
      }
```

...that didn't help.  

I also went as far as removing the `parallelize` call in `test/test_helper.rb` ... still no luck!

I feel like this should be trivial, and yet I'm stumped!  Thanks for the suggestion all the same 👍

**Comment by Syntaf** [comment]
@nadnoslen Np. Are you running rails inside a container or VM? Or is VSCode running within the same environment?

**Comment by nadnoslen** [comment]
Very cool @Syntaf.  I did the same, stopped and started VS Code, no dice.  I even went full _gifsicle_ 👾:

![out](https://user-images.githubusercontent.com/4681678/87616973-a2cd8b00-c6d3-11ea-9e42-a694a588256b.gif)

Showing off a simple `Pundit` policy test, I put several breakpoints in the test class and the instance method that is the subject of the test...alas no breakpoints triggered!

Here's a couple other details about my environment just in case someone else stumbles across this:

```bash
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19] (FROM ASDF-VM)
Rails 6.0.3.2
Using ruby-debug-base 0.2.4.1
Using debase-0.2.4.1

Visual Studio Code 1.47.1

MacOS Catalina 10.15.5
Darwin Macbook-2019.local 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
```

I must be doing something silly.

**Comment by Syntaf** [comment]
@nadnoslen Do you have the `ruby-debug-ide` gem installed? That's what bridges the debugger to VSCode's debugging panel.

**Comment by nadnoslen** [comment]
@Syntaf,

Yes I'm a dummy and copy-pasted stuff poorly...here's an update from `gem list` on my project:

```
ruby-debug-ide (0.7.2)
debase (0.2.4.1)
rails (6.0.3.2)
test-unit (3.3.4)
minitest (5.14.1, 5.13.0)
rack-test (1.1.0)
solargraph (0.39.11)
```

**Comment by Syntaf** [comment]
@nadnoslen Can you add `"showDebuggerOutput": true,` to your debug config and re-run your test? Here's what my debug is showing:

```
Fast Debugger (ruby-debug-ide 0.7.2, debase 0.2.4.1, file filtering is supported) listens on 127.0.0.1:1234

Run options: --seed 42983

# Running:

.

Finished in 0.073565s, 13.5935 runs/s, 13.5935 assertions/s.

1 runs, 1 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by nadnoslen** [comment]
@Syntaf, oh yeah, I think I added and removed that debugger output directive a couple times; here's with it added:

```
Fast Debugger (ruby-debug-ide 0.7.2, debase 0.2.4.1, file filtering is supported) listens on 127.0.0.1:1234

Run options: --seed 13019

# Running:

.

Finished in 0.230894s, 4.3310 runs/s, 12.9930 assertions/s.
1 runs, 3 assertions, 0 failures, 0 errors, 0 skips
```

**Not running in a container at all.  Running locally and natively with `ruby 2.7.1p83`.**

**The debug-enabled Rails server is not running.  And, I don't have rails running anywhere else either.**

Dude, thanks for being so thoughtful and helpful here.  It is truly appreciated.

**Comment by Syntaf** [comment]
@nadnoslen I noticed in your GIF that your test is running within a spring preloader process, that's my last guess lol. Try not using spring for your test during debug and see if that works.

**Comment by nadnoslen** [comment]
@Syntaf, nice work!!! That was actually the problem I was facing.

Updating your supplied launch configuration to include the environment variables allowed me to leave my code untouched:

```json
    {
      "cwd": "${workspaceRoot}",
      "name": "Minitest - Run at cursor position",
      "type": "Ruby",
      "request": "launch",
      "program": "${workspaceRoot}/bin/rails",
      "args": [
        "test",
        "${file}:${lineNumber}"
      ],
      "env": {
        "DISABLE_SPRING": 1,
        "PARALLEL_WORKERS": 1
      },
    }
```

The `DISABLE_SPRING` variable turns off the spring preloader, while the `PARALLEL_WORKERS` set to 1 seems to fix any challenges presented by the `test_helpers` call to `parallelize(workers: :number_of_processors)`.

Hope this helps others.  **Thanks for your wisdom and excellent suggestions!**
